# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1344635975 14400
#      Fri Aug 10 17:59:35 2012 -0400
# Node ID 62a35ae7d6a2666f27a55f7357e5b3b4eae6d218
# Parent  1b2fbc30e4e72c19723534a8281c52c89df6c0fa
use forward decls for mxArray in ov.h and ov-base.h

* ov.h, ov-base.h: Use forward decls for mxArray class instead of
including mxarray.h.
* ov-base-diag.cc, ov-base.cc, ov-base.h, ov-bool-mat.cc, ov-bool.cc,
ov-cell.cc, ov-ch-mat.cc, ov-class.cc, ov-complex.cc, ov-cx-mat.cc,
ov-cx-sparse.cc, ov-float.cc, ov-flt-complex.cc, ov-flt-cx-mat.cc,
ov-flt-re-mat.cc, ov-intx.h, ov-perm.cc, ov-range.cc, ov-re-mat.cc,
ov-re-sparse.cc, ov-scalar.cc, ov-struct.cc, ov-bool-sparse.cc,
ov-perm.cc: Include mxarray.h.
#include "mach-info.h"
* ov-bool-sparse.cc, ov-perm.cc: Include dim-vector.h

diff --git a/src/interp-core/mex.cc b/src/interp-core/mex.cc
--- a/src/interp-core/mex.cc
+++ b/src/interp-core/mex.cc
@@ -30,19 +30,19 @@ along with Octave; see the file COPYING.
 #include <cctype>
 
 #include <set>
 
 #include "f77-fcn.h"
 #include "lo-ieee.h"
 #include "oct-locbuf.h"
 
+#include "Cell.h"
 // mxArray must be declared as a class before including mexproto.h.
-class mxArray;
-#include "Cell.h"
+#include "mxarray.h"
 #include "mexproto.h"
 #include "oct-map.h"
 #include "oct-obj.h"
 #include "ov.h"
 #include "ov-mex-fcn.h"
 #include "ov-usr-fcn.h"
 #include "pager.h"
 #include "parse.h"
diff --git a/src/octave-value/ov-base-diag.cc b/src/octave-value/ov-base-diag.cc
--- a/src/octave-value/ov-base-diag.cc
+++ b/src/octave-value/ov-base-diag.cc
@@ -24,16 +24,17 @@ along with Octave; see the file COPYING.
 #include <config.h>
 #endif
 
 #include <iostream>
 
 #include "mach-info.h"
 #include "lo-ieee.h"
 
+#include "mxarray.h"
 #include "ov-base.h"
 #include "ov-base-mat.h"
 #include "pr-output.h"
 #include "error.h"
 #include "gripes.h"
 #include "oct-stream.h"
 #include "ops.h"
 
diff --git a/src/octave-value/ov-base.cc b/src/octave-value/ov-base.cc
--- a/src/octave-value/ov-base.cc
+++ b/src/octave-value/ov-base.cc
@@ -29,16 +29,17 @@ along with Octave; see the file COPYING.
 
 #include <iostream>
 
 #include "lo-ieee.h"
 #include "lo-mappers.h"
 
 #include "defun.h"
 #include "gripes.h"
+#include "mxarray.h"
 #include "oct-map.h"
 #include "oct-obj.h"
 #include "oct-lvalue.h"
 #include "oct-stream.h"
 #include "ops.h"
 #include "ov-base.h"
 #include "ov-cell.h"
 #include "ov-ch-mat.h"
diff --git a/src/octave-value/ov-base.h b/src/octave-value/ov-base.h
--- a/src/octave-value/ov-base.h
+++ b/src/octave-value/ov-base.h
@@ -27,24 +27,24 @@ along with Octave; see the file COPYING.
 #include <cstdlib>
 
 #include <iosfwd>
 #include <list>
 #include <string>
 
 #include "Range.h"
 #include "data-conv.h"
-#include "mxarray.h"
 #include "mx-base.h"
 #include "str-vec.h"
 
 #include "error.h"
 #include "oct-hdf5.h"
 
 class Cell;
+class mxArray;
 class octave_map;
 class octave_scalar_map;
 class octave_value;
 class octave_value_list;
 class octave_stream;
 class octave_function;
 class octave_user_function;
 class octave_user_script;
diff --git a/src/octave-value/ov-bool-mat.cc b/src/octave-value/ov-bool-mat.cc
--- a/src/octave-value/ov-bool-mat.cc
+++ b/src/octave-value/ov-bool-mat.cc
@@ -29,16 +29,17 @@ along with Octave; see the file COPYING.
 #include <vector>
 
 #include "lo-ieee.h"
 #include "mx-base.h"
 #include "oct-locbuf.h"
 
 #include "defun.h"
 #include "gripes.h"
+#include "mxarray.h"
 #include "oct-obj.h"
 #include "ops.h"
 #include "ov-base.h"
 #include "ov-base-mat.h"
 #include "ov-base-mat.cc"
 #include "ov-bool.h"
 #include "ov-bool-mat.h"
 #include "ov-re-mat.h"
diff --git a/src/octave-value/ov-bool-sparse.cc b/src/octave-value/ov-bool-sparse.cc
--- a/src/octave-value/ov-bool-sparse.cc
+++ b/src/octave-value/ov-bool-sparse.cc
@@ -25,16 +25,19 @@ along with Octave; see the file COPYING.
 #include <config.h>
 #endif
 
 #include <climits>
 
 #include <iostream>
 #include <vector>
 
+#include "dim-vector.h"
+
+#include "mxarray.h"
 #include "ov-base.h"
 #include "ov-scalar.h"
 #include "ov-bool.h"
 #include "ov-bool-mat.h"
 #include "gripes.h"
 #include "ops.h"
 #include "oct-locbuf.h"
 
diff --git a/src/octave-value/ov-bool.cc b/src/octave-value/ov-bool.cc
--- a/src/octave-value/ov-bool.cc
+++ b/src/octave-value/ov-bool.cc
@@ -24,16 +24,17 @@ along with Octave; see the file COPYING.
 #include <config.h>
 #endif
 
 #include <iostream>
 
 #include "mx-base.h"
 
 #include "gripes.h"
+#include "mxarray.h"
 #include "oct-obj.h"
 #include "ops.h"
 #include "ov-bool.h"
 #include "ov-bool-mat.h"
 #include "ov-base.h"
 #include "ov-base-scalar.h"
 #include "ov-base-scalar.cc"
 #include "ov-re-mat.h"
diff --git a/src/octave-value/ov-cell.cc b/src/octave-value/ov-cell.cc
--- a/src/octave-value/ov-cell.cc
+++ b/src/octave-value/ov-cell.cc
@@ -34,16 +34,17 @@ along with Octave; see the file COPYING.
 #include "Array-util.h"
 #include "byte-swap.h"
 #include "lo-utils.h"
 #include "quit.h"
 #include "oct-locbuf.h"
 
 #include "defun.h"
 #include "error.h"
+#include "mxarray.h"
 #include "ov-cell.h"
 #include "oct-obj.h"
 #include "unwind-prot.h"
 #include "utils.h"
 #include "ov-base-mat.h"
 #include "ov-base-mat.cc"
 #include "ov-re-mat.h"
 #include "ov-scalar.h"
diff --git a/src/octave-value/ov-ch-mat.cc b/src/octave-value/ov-ch-mat.cc
--- a/src/octave-value/ov-ch-mat.cc
+++ b/src/octave-value/ov-ch-mat.cc
@@ -26,16 +26,17 @@ along with Octave; see the file COPYING.
 #endif
 
 #include <cctype>
 #include <iostream>
 
 #include "lo-ieee.h"
 #include "mx-base.h"
 
+#include "mxarray.h"
 #include "ov-base.h"
 #include "ov-base-mat.h"
 #include "ov-base-mat.cc"
 #include "ov-ch-mat.h"
 #include "gripes.h"
 #include "pr-output.h"
 
 template class octave_base_matrix<charNDArray>;
diff --git a/src/octave-value/ov-class.cc b/src/octave-value/ov-class.cc
--- a/src/octave-value/ov-class.cc
+++ b/src/octave-value/ov-class.cc
@@ -37,16 +37,17 @@ along with Octave; see the file COPYING.
 #include "error.h"
 #include "file-ops.h"
 #include "gripes.h"
 #include "load-path.h"
 #include "ls-hdf5.h"
 #include "ls-oct-ascii.h"
 #include "ls-oct-binary.h"
 #include "ls-utils.h"
+#include "mxarray.h"
 #include "oct-lvalue.h"
 #include "ov-class.h"
 #include "ov-fcn.h"
 #include "ov-usr-fcn.h"
 #include "pager.h"
 #include "parse.h"
 #include "pr-output.h"
 #include "toplev.h"
diff --git a/src/octave-value/ov-complex.cc b/src/octave-value/ov-complex.cc
--- a/src/octave-value/ov-complex.cc
+++ b/src/octave-value/ov-complex.cc
@@ -25,16 +25,17 @@ along with Octave; see the file COPYING.
 #endif
 
 #include <iostream>
 
 #include "lo-ieee.h"
 #include "lo-specfun.h"
 #include "lo-mappers.h"
 
+#include "mxarray.h"
 #include "oct-obj.h"
 #include "oct-stream.h"
 #include "ops.h"
 #include "ov-complex.h"
 #include "ov-flt-complex.h"
 #include "ov-base.h"
 #include "ov-base-scalar.h"
 #include "ov-base-scalar.cc"
diff --git a/src/octave-value/ov-cx-mat.cc b/src/octave-value/ov-cx-mat.cc
--- a/src/octave-value/ov-cx-mat.cc
+++ b/src/octave-value/ov-cx-mat.cc
@@ -32,16 +32,17 @@ along with Octave; see the file COPYING.
 #include "lo-ieee.h"
 #include "lo-specfun.h"
 #include "lo-mappers.h"
 #include "mx-base.h"
 #include "mach-info.h"
 #include "oct-locbuf.h"
 
 #include "gripes.h"
+#include "mxarray.h"
 #include "oct-obj.h"
 #include "oct-stream.h"
 #include "ops.h"
 #include "ov-base.h"
 #include "ov-base-mat.h"
 #include "ov-base-mat.cc"
 #include "ov-complex.h"
 #include "ov-cx-mat.h"
diff --git a/src/octave-value/ov-cx-sparse.cc b/src/octave-value/ov-cx-sparse.cc
--- a/src/octave-value/ov-cx-sparse.cc
+++ b/src/octave-value/ov-cx-sparse.cc
@@ -29,16 +29,17 @@ along with Octave; see the file COPYING.
 
 #include <iostream>
 #include <vector>
 
 #include "lo-specfun.h"
 #include "lo-mappers.h"
 #include "oct-locbuf.h"
 
+#include "mxarray.h"
 #include "ov-base.h"
 #include "ov-scalar.h"
 #include "ov-complex.h"
 #include "gripes.h"
 
 #include "ov-re-sparse.h"
 #include "ov-cx-sparse.h"
 
diff --git a/src/octave-value/ov-float.cc b/src/octave-value/ov-float.cc
--- a/src/octave-value/ov-float.cc
+++ b/src/octave-value/ov-float.cc
@@ -28,16 +28,17 @@ along with Octave; see the file COPYING.
 
 #include "data-conv.h"
 #include "mach-info.h"
 #include "lo-specfun.h"
 #include "lo-mappers.h"
 
 #include "defun.h"
 #include "gripes.h"
+#include "mxarray.h"
 #include "oct-obj.h"
 #include "oct-stream.h"
 #include "ov-scalar.h"
 #include "ov-float.h"
 #include "ov-base.h"
 #include "ov-base-scalar.h"
 #include "ov-base-scalar.cc"
 #include "ov-flt-re-mat.h"
diff --git a/src/octave-value/ov-flt-complex.cc b/src/octave-value/ov-flt-complex.cc
--- a/src/octave-value/ov-flt-complex.cc
+++ b/src/octave-value/ov-flt-complex.cc
@@ -25,16 +25,17 @@ along with Octave; see the file COPYING.
 #endif
 
 #include <iostream>
 
 #include "lo-ieee.h"
 #include "lo-specfun.h"
 #include "lo-mappers.h"
 
+#include "mxarray.h"
 #include "oct-obj.h"
 #include "oct-stream.h"
 #include "ops.h"
 #include "ov-complex.h"
 #include "ov-base.h"
 #include "ov-base-scalar.h"
 #include "ov-base-scalar.cc"
 #include "ov-flt-cx-mat.h"
diff --git a/src/octave-value/ov-flt-cx-mat.cc b/src/octave-value/ov-flt-cx-mat.cc
--- a/src/octave-value/ov-flt-cx-mat.cc
+++ b/src/octave-value/ov-flt-cx-mat.cc
@@ -32,16 +32,17 @@ along with Octave; see the file COPYING.
 #include "lo-ieee.h"
 #include "lo-specfun.h"
 #include "lo-mappers.h"
 #include "mx-base.h"
 #include "mach-info.h"
 #include "oct-locbuf.h"
 
 #include "gripes.h"
+#include "mxarray.h"
 #include "oct-obj.h"
 #include "oct-stream.h"
 #include "ops.h"
 #include "ov-base.h"
 #include "ov-base-mat.h"
 #include "ov-base-mat.cc"
 #include "ov-complex.h"
 #include "ov-flt-complex.h"
diff --git a/src/octave-value/ov-flt-re-mat.cc b/src/octave-value/ov-flt-re-mat.cc
--- a/src/octave-value/ov-flt-re-mat.cc
+++ b/src/octave-value/ov-flt-re-mat.cc
@@ -37,16 +37,17 @@ along with Octave; see the file COPYING.
 #include "lo-mappers.h"
 #include "mach-info.h"
 #include "mx-base.h"
 #include "quit.h"
 #include "oct-locbuf.h"
 
 #include "defun.h"
 #include "gripes.h"
+#include "mxarray.h"
 #include "oct-obj.h"
 #include "oct-lvalue.h"
 #include "oct-stream.h"
 #include "ops.h"
 #include "ov-base.h"
 #include "ov-base-mat.h"
 #include "ov-base-mat.cc"
 #include "ov-scalar.h"
diff --git a/src/octave-value/ov-intx.h b/src/octave-value/ov-intx.h
--- a/src/octave-value/ov-intx.h
+++ b/src/octave-value/ov-intx.h
@@ -26,16 +26,17 @@ along with Octave; see the file COPYING.
 #include <iosfwd>
 #include <string>
 
 #include "mx-base.h"
 #include "oct-alloc.h"
 #include "str-vec.h"
 
 #include "error.h"
+#include "mxarray.h"
 #include "oct-stream.h"
 #include "ov-base.h"
 #include "ov-base-int.h"
 #include "ov-typeinfo.h"
 #include "gripes.h"
 
 #include "ov-re-mat.h"
 #include "ov-scalar.h"
diff --git a/src/octave-value/ov-perm.cc b/src/octave-value/ov-perm.cc
--- a/src/octave-value/ov-perm.cc
+++ b/src/octave-value/ov-perm.cc
@@ -20,17 +20,19 @@ along with Octave; see the file COPYING.
 
 */
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
 #include "byte-swap.h"
+#include "dim-vector.h"
 
+#include "mxarray.h"
 #include "ov-perm.h"
 #include "ov-re-mat.h"
 #include "ov-scalar.h"
 #include "error.h"
 #include "gripes.h"
 #include "ops.h"
 #include "pr-output.h"
 
diff --git a/src/octave-value/ov-range.cc b/src/octave-value/ov-range.cc
--- a/src/octave-value/ov-range.cc
+++ b/src/octave-value/ov-range.cc
@@ -27,16 +27,17 @@ along with Octave; see the file COPYING.
 #include <iostream>
 
 #include "lo-ieee.h"
 #include "lo-utils.h"
 
 #include "defun.h"
 #include "variables.h"
 #include "gripes.h"
+#include "mxarray.h"
 #include "ops.h"
 #include "oct-obj.h"
 #include "ov-range.h"
 #include "ov-re-mat.h"
 #include "ov-scalar.h"
 #include "pr-output.h"
 
 #include "byte-swap.h"
diff --git a/src/octave-value/ov-re-mat.cc b/src/octave-value/ov-re-mat.cc
--- a/src/octave-value/ov-re-mat.cc
+++ b/src/octave-value/ov-re-mat.cc
@@ -37,16 +37,17 @@ along with Octave; see the file COPYING.
 #include "lo-mappers.h"
 #include "mach-info.h"
 #include "mx-base.h"
 #include "quit.h"
 #include "oct-locbuf.h"
 
 #include "defun.h"
 #include "gripes.h"
+#include "mxarray.h"
 #include "oct-obj.h"
 #include "oct-lvalue.h"
 #include "oct-stream.h"
 #include "ops.h"
 #include "ov-base.h"
 #include "ov-base-mat.h"
 #include "ov-base-mat.cc"
 #include "ov-scalar.h"
diff --git a/src/octave-value/ov-re-sparse.cc b/src/octave-value/ov-re-sparse.cc
--- a/src/octave-value/ov-re-sparse.cc
+++ b/src/octave-value/ov-re-sparse.cc
@@ -29,16 +29,17 @@ along with Octave; see the file COPYING.
 
 #include <iostream>
 #include <vector>
 
 #include "lo-specfun.h"
 #include "lo-mappers.h"
 #include "oct-locbuf.h"
 
+#include "mxarray.h"
 #include "ov-base.h"
 #include "ov-scalar.h"
 #include "gripes.h"
 
 #include "ls-hdf5.h"
 
 #include "ov-re-sparse.h"
 
diff --git a/src/octave-value/ov-scalar.cc b/src/octave-value/ov-scalar.cc
--- a/src/octave-value/ov-scalar.cc
+++ b/src/octave-value/ov-scalar.cc
@@ -28,16 +28,17 @@ along with Octave; see the file COPYING.
 
 #include "data-conv.h"
 #include "mach-info.h"
 #include "lo-specfun.h"
 #include "lo-mappers.h"
 
 #include "defun.h"
 #include "gripes.h"
+#include "mxarray.h"
 #include "oct-obj.h"
 #include "oct-stream.h"
 #include "ov-scalar.h"
 #include "ov-float.h"
 #include "ov-base.h"
 #include "ov-base-scalar.h"
 #include "ov-base-scalar.cc"
 #include "ov-re-mat.h"
diff --git a/src/octave-value/ov-struct.cc b/src/octave-value/ov-struct.cc
--- a/src/octave-value/ov-struct.cc
+++ b/src/octave-value/ov-struct.cc
@@ -25,16 +25,17 @@ along with Octave; see the file COPYING.
 #endif
 
 #include <iostream>
 
 #include "Cell.h"
 #include "defun.h"
 #include "error.h"
 #include "gripes.h"
+#include "mxarray.h"
 #include "oct-lvalue.h"
 #include "ov-struct.h"
 #include "unwind-prot.h"
 #include "utils.h"
 #include "variables.h"
 
 #include "Array-util.h"
 #include "oct-locbuf.h"
diff --git a/src/octave-value/ov.h b/src/octave-value/ov.h
--- a/src/octave-value/ov.h
+++ b/src/octave-value/ov.h
@@ -29,26 +29,26 @@ along with Octave; see the file COPYING.
 #include <iosfwd>
 #include <string>
 #include <list>
 
 #include "Range.h"
 #include "data-conv.h"
 #include "idx-vector.h"
 #include "mach-info.h"
-#include "mxarray.h"
 #include "mx-base.h"
 #include "oct-alloc.h"
 #include "oct-time.h"
 #include "str-vec.h"
 
 #include "oct-hdf5.h"
 #include "oct-sort.h"
 
 class Cell;
+class mxArray;
 class octave_map;
 class octave_scalar_map;
 class Octave_map;
 class octave_stream;
 class octave_function;
 class octave_user_function;
 class octave_fcn_handle;
 class octave_fcn_inline;
